### PR TITLE
50: Live mic monitoring — silent mic detection and per-heartbeat level stats in record_mics.py

### LINKED ISSUE #50: S22: Live mic monitoring — silent mic detection and per-heartbeat level stats in record_mics.py
## Context

During today's recording session (`panama-hortigal-estate-roast3`), mic2 (ATR2100x) was physically off and produced no signal. There was no feedback from the tool — the session completed without warning, resulting in a silent second channel and an unusable paired recording.

Additionally, there is no way to tell during a live session whether the mics are balanced or capturing meaningful audio. The only way to check is to stop, inspect post-hoc, and re-record.

## Goal

Three improvements to `record_mics.py` to provide real-time mic health feedback:

---

## 1. Silent mic detection

After the first `N` seconds of audio have buffered (e.g. ~5s), analyse each mic channel and warn if the signal is effectively silent:

- RMS below a threshold (e.g. `-60 dBFS`) → warn once per mic: `⚠️  mic2 (audiotechnica): no signal detected — is it turned on?`
- Re-check periodically (e.g. every 30s) and clear the warning if signal recovers
- Do not block or abort — just print to stderr

## 2. Per-heartbeat live stats

The existing 30-second heartbeat line:
```
[MM:SS] Recording...
```

Should include per-mic peak/RMS and a balance indicator, e.g.:
```
[10:32] mic1: peak=-19.2 dBFS rms=-37.1 dBFS | mic2: peak=-22.1 dBFS rms=-44.3 dBFS | balance=7.2dB ⚠️
[10:32] mic1: peak=-19.2 dBFS rms=-37.1 dBFS | mic2: peak=-21.8 dBFS rms=-43.9 dBFS | balance=6.8dB ✅
```

Stats are computed over the audio captured **since the last heartbeat** (i.e. the most recent 30s window), not the full session, to reflect current conditions.

Balance warning threshold: > 6 dB RMS difference between any two mics.

## 3. Final summary after writing

After all WAV files are written, print a per-mic level summary and balance check:

```
--- Recording summary ---
mic1 (fifine):        peak=-17.2 dBFS  rms=-36.1 dBFS  ✅
mic2 (audiotechnica): peak=-21.3 dBFS  rms=-44.0 dBFS  ✅
Balance: 7.9 dB  ⚠️  UNBALANCED (>6dB)
```

This is computed from the full session audio (same data used for WAV writes).

## Acceptance Criteria

- [ ] Silent mic warning fires within ~10s of session start if any mic RMS < -60 dBFS
- [ ] Warning is printed to stderr and does not interrupt the recording
- [ ] Heartbeat line extended with per-mic peak/RMS and balance for the last 30s window
- [ ] Balance warn symbol shown when RMS difference > 6 dB
- [ ] Final summary printed after WAV writes with full-session peak/RMS per mic + balance
- [ ] `ruff check`, `ruff format`, `pyright` pass
- [ ] Existing tests pass; new tests cover silent-mic detection and summary output

## Links
- Epic: #1
- Depends on: #46 (`record_mics.py` base)
- Observed during: `panama-hortigal-estate-roast3` (2026-04-12)

### PR DESCRIPTION
**Pull request by syamaner**
## What this PR contains

Implements three live monitoring improvements to `scripts/record_mics.py` (issue #50).

### 1. Silent mic detection

Within ~5 seconds of recording start, each mic channel is checked for signal. If any mic's RMS falls below -60 dBFS, a warning is printed to stderr:

```
⚠️  mic2 (audiotechnica): no signal detected (rms=-78.7 dBFS) — is it turned on?
```

The check re-runs at every 30-second heartbeat. Each mic is warned at most once (no double-warn). Fires regardless of `--quiet` since it goes to stderr.

**Motivation**: today's first `panama-hortigal-estate-roast3` session was lost because mic2 was physically off — no feedback until post-hoc balance check.

### 2. Per-heartbeat live stats

The 30s heartbeat line now shows per-mic peak/RMS for the **last 30-second window** and an inline balance indicator:

```
[10:32] mic1(fifine): peak=-19.2 rms=-37.1 dBFS | mic2(audiotechnica): peak=-21.8 rms=-43.9 dBFS | balance=6.8dB ✅
[10:32] mic1(fifine): peak=-19.2 rms=-37.1 dBFS | mic2(audiotechnica): peak=-63.3 rms=-78.7 dBFS | balance=39.4dB ⚠️
```

Balance warning (`⚠️`) fires when RMS difference between any two mics exceeds 6 dB.

### 3. Full-session summary after writing

After all WAV files are written, a complete level report is printed:

```
--- Recording summary ---
  mic1 (fifine        ):  peak= -17.2 dBFS   rms= -36.1 dBFS
  mic2 (audiotechnica ):  peak= -21.3 dBFS   rms= -44.0 dBFS
  Balance: 7.9 dB  ⚠️  UNBALANCED (>6dB)
```

Flags: `⚠️  CLIPPING` (peak ≥ -0.5 dBFS), `⚠️  TOO QUIET` (peak < -30 dBFS).

---

## New helpers (all tested)

- `_dbfs(arr)` — peak + RMS dBFS from a 1-D float32 array
- `_mic_stats_from_chunks(chunks, mics, gains, start_chunk)` — per-mic stats with optional windowing
- `_format_heartbeat(elapsed_sec, stats, mics, labels)` — formats the heartbeat line
- `_check_silent_mics(stats, mics, labels, warned)` — warns once per mic, returns updated warned set
- `_print_session_summary(stats, mics, labels)` — final level report

## Other changes

- `tests/test_record_mics.py` — 35 new tests, 51/51 passing
- `.claude/skills/record-roast/SKILL.md` — new skill documenting recorder usage; updated with live monitoring behaviour
- `docs/multi_mic_setup.md` — minor doc updates

## Checks

- `ruff check` + `ruff format` ✅
- `pytest` 51/51 ✅

Closes #50

Co-Authored-By: Oz <oz-agent@warp.dev>

### CODE REVIEW

File: scripts/record_mics.py [hunk 1]
```diff
@@ -266,13 +435,36 @@ def _callback(
...
+                # Fires once regardless of --quiet so the warning always reaches stderr.
+                if not silence_checked and elapsed >= _SILENCE_CHECK_AFTER_SEC:
+                    with lock:
+                        init_chunks = list(chunks)
+                    if init_chunks:
+                        init_stats = _mic_stats_from_chunks(init_chunks, mics, gains)
+                        silence_warned = _check_silent_mics(
+                            init_stats, mics, labels, silence_warned
+                        )
+                    silence_checked = True
```
[the remark this answers is not in the document]
Comment (syamaner) [reply scripts/record_mics.py]: Fixed in `c87cb99`. The initial silence check now only completes after at least one chunk has been captured, so startup callback latency no longer suppresses the early silent-mic warning. I also added a regression test for the empty-buffer retry case.
